index.html

```diff
@@ -2876,13 +2876,27 @@ <h3><dfn>Writing or pushing content</dfn></h3>
             Let |language:string| be |record|'s <a>lang</a> if it [= map/exists =],
             or else to |documentLanguage|.
           </li>
-          <li>
-            Let |encoding label:string| be |record|'s <a>encoding</a> if it
-            [= map/exists =], or "`utf-8`".
-          </li>
-          <li>
-            If |encoding label| is not equal to "`utf-8`", "`utf-16`",
-            "`utf-16le`" or "`utf-16be`" [= exception/throw =] a {{TypeError}}.
+          <li>Switch on the type of |record|'s <a>data</a>:
+            <dl>
+              <dt>{{DOMString}}</dt>
+              <ol>
+                <li>
+                  Let |encoding label:string| be "`utf-8`".
+                </li>
+              </ol>
+              <dt>{{BufferSource}}</dt>
+              <ol>
+                <li>
+                  Let |encoding label:string| be |record|'s <a>encoding</a> if
+                  it [= map/exists =], or "`utf-8`".
+                </li>
+                <li>
+                  If |encoding label| is not equal to "`utf-8`", "`utf-16`",
+                  "`utf-16le`" or "`utf-16be`" [= exception/throw =] a
+                  {{TypeError}}.
```

Comment:
TypeError seems wrong as it is not something you set...

Reply:
Oh it is, probably makes sense to move this test before the other then

Reply:
Not sure what you mean. Which other test?

Reply:
Not test, I meant line

Reply:
How? This looks good to me.
 
![image](https://user-images.githubusercontent.com/634478/71978545-e5da0d80-321b-11ea-9237-d4607228c668.png)

Reply:
yeah its fine.